src/main/kotlin/org/ptss/support/security/IdentityServiceClient.kt

```diff
@@ -0,0 +1,49 @@
+package org.ptss.support.security
+
+import io.quarkus.security.UnauthorizedException
+import io.smallrye.jwt.build.Jwt
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import org.ptss.support.domain.contants.TokenExpiration
+import org.ptss.support.domain.enums.Role
+import java.util.Date
+
+@ApplicationScoped
+class IdentityServiceClient @Inject constructor(
+    private val jwtValidator: JwtValidator
+) {
+    fun refreshAccessToken(refreshToken: String): String {
+        val newToken = getNewAccessToken(refreshToken)
+        if (!jwtValidator.isTokenValidAndNotBlank(newToken)) {
+            throw UnauthorizedException("Failed to obtain valid access token")
+        }
```

Comment:
would throw here our custom apiexception, as it isn't really a unauthorized exception
```
        throw APIException(
            errorCode = ErrorCode.TOKEN_GENERATION_FAILED,
            message = "Failed to generate valid access token"
        )
```